### PR TITLE
A couple fixes related to complex data types in Iceberg connector

### PR DESCRIPTION
**Pull request by anjalinorwood**

### CODE REVIEW

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -539,14 +539,16 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
     private static Schema toIcebergSchema(List<ColumnMetadata> columns)
     {
         List<NestedField> icebergColumns = new ArrayList<>();
+        int index = 1;
```
Comment (electrum) [inline presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java:542]: Have you considered making `TypeConverter` an object that stores the index? That should be simpler than passing it through static methods with recursive invocations. Something like
```java
TypeConverter converter = new TypeConverter();
for (ColumnMetadata column : columns) {
    Type type = converter.toIcebergType(column.getType());
    int index = converter.getIndex();
    ...
}
```
We'd probably want to split up the class into `PrestoTypeConverter` and `IcebergTypeConverter` as a precursor commit, since `toPrestoType` can stay static.

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import io.prestosql.tempto.AfterTestWithContext;
+import io.prestosql.tempto.BeforeTestWithContext;
+import io.prestosql.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.CREATE_TABLE;
+
+public class TestCreateTable
```
Comment (electrum) [inline presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java:26]: Can you add these tests to `TestIcebergSmoke`? Product tests should only be used if absolutely necessary, since they are harder to run (you can't run them in the IDE or easily debug them).

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    @BeforeTestWithContext
+    public void createObjects()
+    {
+        // Drop Iceberg tables and schema
+        try {
+            query("DROP TABLE IF EXISTS " + ICEBERG_TABLE_NAME1);
+            query("DROP TABLE IF EXISTS " + ICEBERG_TABLE_NAME2);
+        }
+        catch (Exception e) {
```
Comment (electrum) [inline presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java:41]: We shouldn't catch exceptions like this in tests, as we want any such exception to fail the test.

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TypeConverter.java [hunk 4]
```diff
@@ -199,7 +199,7 @@ private static PrestoToIcebergReturnDataType fromRow(RowType type, int index)
             String name = field.getName().orElseThrow(() ->
                     new PrestoException(NOT_SUPPORTED, "Row type field does not have a name: " + type.getDisplayName()));
             PrestoToIcebergReturnDataType returnType = toIcebergType(field.getType(), index);
-            fields.add(Types.NestedField.required(index, name, returnType.getType()));
+            fields.add(Types.NestedField.optional(index, name, returnType.getType()));
```
Comment (electrum) [inline presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TypeConverter.java:202]: Can you add a test to `TestIcebergSmoke` that creates a table with a row type, inserts a null into one of the row fields, then reads it?

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java (hunk 1 above)
Comment (anjalinorwood) [reply presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java:542]: I made these changes as you suggested. Turns out there is another commit that is not yet pushed upstream that requires this fix. So, I am going to abandon this particular commit and upload a new one when the other commit is pushed upstream (soon).

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java (hunk 2 above)
Comment (anjalinorwood) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java:26]: Yes, moved the tests to TestIcebergSmoke. 
I was able to run product tests in IDE similar to how I run smoke, but I agree that smoke tests is a better place for these.

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java (hunk 3 above)
Comment (anjalinorwood) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestCreateTable.java:41]: Got it.

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TypeConverter.java (hunk 4 above)
Comment (anjalinorwood) [reply presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TypeConverter.java:202]: Yes.

**Comment by electrum** [comment]
Thanks!
